### PR TITLE
Update postgres_test.go

### PR DESCRIPTION
**Pull request by leetal**
Updated to use dockertest V2.

### CODE REVIEW

File: storage/postgres/postgres_test.go [hunk 1]
```diff
@@ -23,20 +23,36 @@ var store *Storage
...
+		db = dbTemp
+
+		return db.Ping() == nil
+	})
+
 	if err != nil {
-		log.Fatalf("Could not set up PostgreSQL container: %v", err)
+		log.Fatalf("Could not connect to database: %s", err)
 	}
-	defer c.KillRemove()
```
Comment (arekkas) [inline storage/postgres/postgres_test.go]: Why did you remove `defer c.KillRemove`? `defer` runs always on function exit and is therefore better suited than calling `tearDown` manually

File: storage/postgres/postgres_test.go [hunk 2]
```diff
@@ -23,20 +23,36 @@ var store *Storage
...
-	}
+	retCode := m.Run()
+
+	// force teardown
+	tearDown(c)
+
+	os.Exit(retCode)
+}
 
-	os.Exit(m.Run())
```
Comment (arekkas) [inline storage/postgres/postgres_test.go]: We don't need `retCode` so I think we can keep this line

Comment (leetal) [inline storage/postgres/postgres_test.go]: The defer is never called due to os.Exit() being called before
c.KillRemove() is run. Test it locally, the docker image is not removed
(not on mac at least). By actually forcing a teardown finction to be
called, we can therefore force the c.KillRemove() to be called always, in
all cases.

Comment (arekkas) [inline storage/postgres/postgres_test.go]: Oh I see so `os.Exit` prevents calling `defer` because go terminates instantly? Makes sense and explains why my docker machine is always full...

Comment (leetal) [inline storage/postgres/postgres_test.go]: Yep :) Basically so ;)

I 'm currently fixing all other tests as well to use your new dockertest
V2. But something strange is happening on the Travis builds, since it seems
nothing can connect to postgres?! RethinkDB is working fine?

Locally everything works flawlessly, but not on travis for some reason?
Perhaps different postgres images is used on travis contra locally?

Comment (arekkas) [inline storage/postgres/postgres_test.go]: Yeah really weird. I'm looking into it right now

Comment (arekkas) [inline storage/postgres/postgres_test.go]: Oh ok the problem is here:

```
	c, err := dockertest.ConnectToPostgreSQL(15, time.Second, func(url string) bool {
		var err error
		dbTemp, err := sql.Open("postgres", url)
		if err != nil {
			return false
		}
		store = New(dbTemp)
		if err = store.CreateSchemas(); err != nil {
			log.Fatalf("Could not ping database: %v", err)
		}

		db = dbTemp

		return db.Ping() == nil
	})
```

The callback method is just for checking if the database is already responsive. Depending on the environment (e.g. travis VM), docker images need a little time before they go responsive - although they are booted already! An unresponsive docker image (e.g. postgres cleaning up -> then listening) is solved by checking if the database is pingable. If it's not, dockertest retries connecting.

Therefore, CreateSchemas must not be inside of that callback function but instead below it. I will fix this and ping you in the PR.

My mistake for not catching that during review.

Comment (leetal) [inline storage/postgres/postgres_test.go]: Ah, okey! Good :)

Comment (arekkas) [inline storage/postgres/postgres_test.go]: It's now documented [here](https://github.com/ory-am/dockertest#start-a-container) as well :)

Comment (leetal) [inline storage/postgres/postgres_test.go]: Awesome!
I really missed that "retry" part, but it was 10x more obvious that you had
it in there when you commented about the delays ;)
